main.cpp

```diff
@@ -43,18 +44,78 @@ struct Point
     float x{0}, y{0};
 };
 
+// --------------------------------------------------
+//               Wrapper class templates
+// --------------------------------------------------
+
+// Templated Wrapper struct
 template<typename Type>
 struct Wrapper
 {
     Wrapper(Type&& t) : val(std::move(t)) 
     { 
+        std::cout << "TypeID: "; 
         std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
     }
+    void print() 
+    {
+        std::cout << "Value: "; 
+        std::cout << "Wrapper::print(" << val << ")" << std::endl; 
+        std::cout << "------------------------" << std::endl; 
+    }    
+
+    Type val;
+
 };
 
-int main()
+// Templated specialization of Wrapper struct for Point
+template<>
+struct Wrapper<Point>
 {
-    variadicHelper( 3, std::string("burgers"), 2.5, Point{3.f, 0.14f} );
+    Wrapper(Point&& t) : val(std::move(t)) 
+    { 
+        std::cout << "TypeID: "; 
+        std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
+    }
+    void print() 
+    { 
+        std::cout << "Value: "; 
+        std::cout << "Wrapper::print(" << val.toString() << ")" << std::endl; 
+        std::cout << "------------------------" << std::endl; 
+    }
+
+    Point val;
+};
+
+// --------------------------------------------------
+//               Variadic templates
+// --------------------------------------------------
+
+// Single-template-parameter version
+template<typename T>
+void variadicHelper(T first)
+{
```

Comment:
You're forcing a pass-by-copy.
- [ ] fix your parameter type.  see instruction 9 in instructions.cpp